MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/offline/CacheManagementActivity.java

```diff
@@ -0,0 +1,284 @@
+    super.onCreate(savedInstanceState);
+
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_offline_cache_manipulation);
+
+    // Get an instance of Firebase Remote Config.
+    firebaseRemoteConfig = FirebaseRemoteConfig.getInstance();
+
+    // Sets the minimum interval between successive fetch calls. Fetches less than
+    // number of seconds passed in as a parameter, after the last fetch from the Firebase Remote
+    // Config server would use values returned during the last fetch.
+    // More info at https://bit.ly/2XJdvuT.
+
+    // NOT recommended for apps in production. Uncomment to use for app development and testing
+    /*firebaseRemoteConfig.setConfigSettingsAsync(new FirebaseRemoteConfigSettings.Builder()
+        .setMinimumFetchIntervalInSeconds(5)
+        .build());*/
+
+    // Set default Remote Config parameter values. The app uses the in-app default values, and
+    // when you need to adjust those defaults, you set an updated value for only the values you
+    // want to change in the Firebase console.
+    firebaseRemoteConfig.setDefaults(R.xml.cache_management_remote_config_defaults);
+
+    // Fetch the latest Remote Config values from Firebase and then adjust the cache if needed.
+    firebaseRemoteConfig.fetchAndActivate().addOnCompleteListener(this,
+        new OnCompleteListener<Boolean>() {
+          @Override
+          public void onComplete(@NonNull Task<Boolean> task) {
+            if (task.isSuccessful()) {
+              adjustCacheBasedOnRemoteConfigServerSettings();
+            } else {
+              Logger.d("CacheManagementActivity", "Couldn\'t get Firebase cache "
+                  + "management remote config settings from Firebase.");
+            }
+          }
+        });
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull MapboxMap mapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS, new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            findViewById(R.id.invalidate_ambient_cache_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                invalidateAmbientCache();
+              }
+            });
+
+            findViewById(R.id.set_max_ambient_cache_size_button).setOnClickListener(new View.OnClickListener() {
+              @Override
+              public void onClick(View view) {
+                EditText newMaxAmbientCacheSizeTextView = findViewById(
```

Comment:
NIT: `findViewById` calls are generally expensive and shouldn't be used in recurring scenarios (button click). They should rather be cached.

Reply:
Fixed